apps/mull-ui/src/app/pages/create-event/create-event.tsx

```diff
@@ -0,0 +1,205 @@
+    date.setMinutes(parseInt(minute));
+  };
+
+  const notifySubmissionToast = () => {
+    toastId.current = toast('Submitting Event...', { autoClose: false });
+  };
+
+  const updateSubmissionToast = (type: TypeOptions, message: string) => {
+    toast.update(toastId.current, {
+      type,
+      render: message,
+      autoClose: 3000,
+    });
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      activeRestriction: EventRestriction.NONE,
+      startDate: null,
+      endDate: null,
+      startTime: '',
+      endTime: '',
+      eventTitle: '',
+      description: '',
+      location: '',
+    },
+
+    validationSchema: Yup.object({
+      startDate: Yup.date().required('Start date is required'),
+      endDate: Yup.date()
+        .nullable()
+        .required('End date is required')
+        .test('maxEventLength', 'Event length cannot be over 30 days.', function (endDate) {
+          const diff = Math.abs(+endDate - +this.parent.startDate);
+          return diff <= 30 * DAY_IN_MILLISECONDS;
+        }),
+      startTime: Yup.string().required('Start Time is required.'),
+      endTime: Yup.string().required('End Time is required.'),
+      eventTitle: Yup.string()
+        .required('Event Title is required.')
+        .max(65, 'Event Title length must be under 65 characters.'),
+      activeRestriction: Yup.number().min(0).max(2),
+      description: Yup.string()
+        .required('Event Description is required.')
+        .max(5000, 'Event Description must be under 5000 characters.'),
+      location: Yup.string().required('Event Location is required.'),
+    }),
+
+    onSubmit: (values) => {
+      notifySubmissionToast();
+      addTimeToDate(values.startTime, values.startDate);
+      addTimeToDate(values.endTime, values.endDate);
+      const payload = {
+        startDate: values.startDate,
+        endDate: values.endDate,
+        description: values.description,
+        title: values.eventTitle,
+        restriction: values.activeRestriction,
+      };
+      createEvent({ variables: { createEventInput: payload } })
+        .then(({ errors }) => {
+          if (errors) {
+            updateSubmissionToast(toast.TYPE.ERROR, 'Event Not Created');
+          } else {
+            updateSubmissionToast(toast.TYPE.SUCCESS, 'Event Created');
+            history.push('/home');
+          }
+        })
+        .catch(() => {
+          updateSubmissionToast(toast.TYPE.ERROR, 'Fatal Error: Event Not Created');
+        });
+    },
+  });
+
+  const handleRestrictions = (idx: number) => {
+    formik.setFieldValue('activeRestriction', idx);
+  };
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <div className="create-event page-container">
+        <p className="create-event-text">Create Event</p>
+        <label htmlFor="imageFile" className="custom-file-upload event-input-border">
+          {imageFile ? (
+            <img src={imageFile} style={{ width: '50%', height: '50%' }} alt="sup" />
```

Comment:
What should they do instead? I had a related comment

Reply:
The label is being used here to be able to click the hidden input. What I've done in other projects is have a reference (with useRef hook) to the hidden input and when the parent div is clicked it calls the .click() function on the reference which is the input.

Since this PR doesn't deal with image uploading I'm okay with leaving the updating of the UI to #28. Just center the image to match the mockup.

Reply:
That's a good way of doing it yeah, if @JimLau49 and @bit172 don't want to tackle this now we can do it in #28. Feel free to use the code I suggested in my comment to center the image and remove the padding to better match the mockup.